app/assets/javascripts/uploadAdditionalInformation.js

```diff
@@ -0,0 +1,84 @@
+$(document).ready(function(){
+	Dropzone.autoDiscover = false;
+
+	// Source: https://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript#answer-2117523
+	function uuidv4() {
+		return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+			(c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
+		)
+	}
+
+	function setUploadUrl(url) {
+		uploadAdditionalInfoDropzone.options.url = url;
+	}
+
+	function prepareAwsHiddenInputs(responseData) {
+		$('#aws_key').attr('name', 'key').val(responseData.fields['key']);
+		$('#aws_policy').attr('name', 'policy').val(responseData.fields['policy']);
+		$('#aws_x-amz-algorithm').attr('name', 'x-amz-algorithm').val(responseData.fields['x-amz-algorithm']);
+		$('#aws_x-amz-credential').attr('name', 'x-amz-credential').val(responseData.fields['x-amz-credential']);
+		$('#aws_x-amz-date').attr('name', 'x-amz-date').val(responseData.fields['x-amz-date']);
+		$('#aws_x-amz-signature').attr('name', 'x-amz-signature').val(responseData.fields['x-amz-signature']);
+		$('#aws_success_action_status').attr('name', 'success_action_status');
+	}
+
+	function getPresignedS3Url (cb) {
+		$.ajax({
+			url: '/api/s3/create_signed_url',
+			dataType: 'json',
+			data: {
+					uuid: uuidv4(),
+					command: 'CreateSignedS3FormData',
+					async: false,
+					data: {
+					    rubbish: '1'
+                    }
+			},
+			error: onGetPresignedError,
+			method: 'post',
+			success: function (data, status, xhr) {
+				cb.apply(this, [data.data]);
+			}
+		});
+	}
+
+	function onGetPresignedError(xhr, status, error) {
+		/* TODO: Anticipate and handle errors:
+			- Network issue to API (no response)
+			- Network issue to/from S3 (API responds with bad data)
+		*/
+	}
+
+	uploadAdditionalInfoDropzone = new Dropzone("#upload-additional-file", 
```

Comment:
Have you ran this through a linter ?  Should this not have a 'var' before it ?

Reply:
Tidied up in 2a49dd0